src/datachain/catalog/catalog.py

```diff
@@ -1218,6 +1218,10 @@ def register_dataset(
             preview=dataset_version.preview,
             job_id=dataset_version.job_id,
         )
+
+        session = Session.get_current_context()
```

Comment:
`session` can be also passed to DataChain explicitly (check `from_storage`, for example) ... it's probably expected that we pass it directly into all the datacnain factories even when we use contextmanager ... not sure `get_current_context` is the right approach ... at least it might be changing the assumption ... can we check docs / existing tests please?

Reply:
Updated the logic along with example in the file where the variable session is used. PTAL.